src/overlays/actors/ovl_En_Viewer/z_en_viewer.c

```diff
@@ -59,144 +62,169 @@ static InitChainEntry sInitChain[] = {
-    { OBJECT_ZL4, OBJECT_OPENING_DEMO1, 1, 0, 0, 10, 2, &gChildZeldaSkel, &object_opening_demo1_Anim_000450 },
-    { OBJECT_GNDD, OBJECT_GNDD, 1, -6, 0, 10, 0, &object_gndd_Skel_0119E8, &object_gndd_Anim_002928 },
-    { OBJECT_HORSE_GANON, OBJECT_HORSE_GANON, 1, 0, 2, 20, 1, &gHorseGanonSkel, &gHorseGanonRearingAnim },
-    { OBJECT_GNDD, OBJECT_GNDD, 1, -6, 0, 10, 0, &object_gndd_Skel_0119E8, &object_gndd_Anim_0005B4 },
-    { OBJECT_HORSE_GANON, OBJECT_HORSE_GANON, 1, 0, 2, 20, 1, &gHorseGanonSkel, &gHorseGanonGallopingAnim },
-    { OBJECT_GNDD, OBJECT_GNDD, 1, -6, 0, 10, 0, &object_gndd_Skel_0119E8, &object_gndd_Anim_004260 },
-    { OBJECT_GNDD, OBJECT_GNDD, 1, -6, 0, 10, 0, &object_gndd_Skel_0119E8, &object_gndd_Anim_0050A8 },
-    { OBJECT_GANON, OBJECT_GANON, 1, -6, 0, 10, 0, &object_ganon_Skel_0114E8, &object_ganon_Anim_011348 },
+// clang-format off
+static EnViewerInitData sInitData[] = {
+    /* 0 */ { OBJECT_HORSE_ZELDA, OBJECT_HORSE_ZELDA, 1, 0,
+              ENVIEWER_SHADOW_HORSE, 20, ENVIEWER_DRAW_HORSE,
+              &gHorseZeldaSkel, &gHorseZeldaGallopingAnim },
+    /* 1 */ { OBJECT_IM, OBJECT_OPENING_DEMO1, 1, 0,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_IMPA,
+              &object_im_Skel_00F788, &object_opening_demo1_Anim_0029CC },
+    /* 2 */ { OBJECT_ZL4, OBJECT_OPENING_DEMO1, 1, 0,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_ZELDA,
+              &gChildZeldaSkel, &object_opening_demo1_Anim_000450 },
+    /* 3 */ { OBJECT_GNDD, OBJECT_GNDD, 1, -6,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_GANON,
+              &object_gndd_Skel_0119E8, &object_gndd_Anim_002928 },
+    /* 4 */ { OBJECT_HORSE_GANON, OBJECT_HORSE_GANON, 1, 0,
+              ENVIEWER_SHADOW_HORSE, 20, ENVIEWER_DRAW_HORSE,
+              &gHorseGanonSkel, &gHorseGanonRearingAnim },
+    /* 5 */ { OBJECT_GNDD, OBJECT_GNDD, 1, -6,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_GANON,
+              &object_gndd_Skel_0119E8, &object_gndd_Anim_0005B4 },
+    /* 6 */ { OBJECT_HORSE_GANON, OBJECT_HORSE_GANON, 1, 0,
+              ENVIEWER_SHADOW_HORSE, 20, ENVIEWER_DRAW_HORSE,
+              &gHorseGanonSkel, &gHorseGanonGallopingAnim },
+    /* 7 */ { OBJECT_GNDD, OBJECT_GNDD, 1, -6,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_GANON,
+              &object_gndd_Skel_0119E8, &object_gndd_Anim_004260 },
+    /* 8 */ { OBJECT_GNDD, OBJECT_GNDD, 1, -6,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_GANON,
+              &object_gndd_Skel_0119E8, &object_gndd_Anim_0050A8 },
+    /* 9 */ { OBJECT_GANON, OBJECT_GANON, 1, -6,
+              ENVIEWER_SHADOW_NONE, 10, ENVIEWER_DRAW_GANON,
+              &object_ganon_Skel_0114E8, &object_ganon_Anim_011348 },
 };
+// clang-format on
 
-static EnViewerAnimFunc sAnimFuncs[] = {
-    func_80B2A300,
-    func_80B2A4D8,
-    func_80B2A300,
-    func_80B2A448,
+static EnViewerInitDrawFunc sInitDrawFuncs[] = {
+    EnViewer_InitDrawGanonOrZelda,
```

Comment:
not too sure i agree with "InitDraw". i dont consider animation stuff really as a part of draw. it seems mostly concerned with skeletons and animations. So I would probably go with `InitSkelAnim` (without the "e", i mean initializing skeleton and animation, not the system itself lol)